modules/View.js

```diff
@@ -154,32 +158,130 @@
+     * @param newView {View} The instantiated view object to be transitioned into the injection site
+     * @param [options] {Object} optional options object. This options object will be passed on to the transitionIn and transitionOut methods as well.
+     * @param   [options.previousView] {View} the view that should be transitioned out. If none is provided, it will look to see if a view already
+     *                                 is at this injection site and uses that by default.
+     * @param   [options.addBefore=false] {Boolean} if true, the new view's element will be added before the previous view's element. Defaults to after.
+     * @param   [options.shared=false] {Boolean} if set to true, the view will be treated as a shared view and not disposed during parent view disposing.
+     * @return {Promise} resolved when all transitions are complete. No payload is provided upon resolution.
+     * When the transitionIn and transitionOut methods are invoked on the new and previous views, the options parameter will be passed on to them. Other fields
+     * will be added to the options parameter to allow better handling of the transitions. These include:
+     * {
+     *   newView: the new view
+     *   previousView: the previous view (can be undefined)
+     *   parentView: the parent view transitioning in or out the tracked view
+     * }
      */
-    injectView: function(injectionSite, view, options) {
-      var injectionPoint = this.$('[inject=' + injectionSite + ']');
-      if (view && injectionPoint.size() > 0) {
-        this.attachView(injectionPoint, view, options);
+    transitionNewViewIntoSite: function(injectionSiteName, newView, options) {
+      var previousView, injectionPoint, newInjectionSite, currentPromise,
+        previousDeferred = $.Deferred();
+      options = options || {};
+      // find previous view that used this injection site.
+      previousView = options.previousView;
+      if (!previousView) {
+        previousView = this.__getLastTrackedViewAtInjectionSite(injectionSiteName);
+      }
+      _.defaults(options, {
+        parentView: this,
+        newView: newView,
+        previousView: previousView
+      });
+      options.useTransition = false;
+      if (previousView == newView) {
+        // Inject this view like normal if it's already the last one there
+        return this.injectView(injectionSiteName, newView, options);
+      }
+      if (!previousView) {
+        // Only transition in the new current view and find the injection point.
+        injectionPoint = this.$('[inject=' + injectionSiteName + ']');
+        return this.transitionInView(injectionPoint, newView, options);
+      }
+      this.injectView(injectionSiteName, previousView, options);
+      options.cachedInjectionSite = previousView.injectionSite;
+      newInjectionSite = options.newInjectionSite = $('<div inject="' + injectionSiteName + '">');
```

Comment:
We can definietly make it a span.

Here's the process:
We re-render like normally, injecting in the old view, then adding an equivolent injection site next to the old view, finally transitioning in the new view into the new injection site.

The reason I did this instead of just plopping the new view in "before" or "after" the old view without an injection site, is because this way we get to reuse the same logic/methods to transition in the new view (using an injection site). Plus, the new view needs a stored injection site anyway (so that it can be transitioned out later). It made sense to follow the same pattern.